### ISSUE TITLE
False positive: URL inside HTML <a> tag flagged as bare URL by no-bare-urls rule

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#336]
## Problem

A URL used as link text inside an `<a>` tag is incorrectly flagged as a bare URL by the `no-bare-urls` rule.

## Reproduction

The following HTML triggers the error:

```html
<a href="https://rustup.rs" target="_blank" rel="noopener noreferrer">https://rustup.rs</a>
```

Error message:

```
bare URL found, use angle brackets or a Markdown link: https://rustup.rs
```

## Expected behavior

URLs inside `<a>` tags should not be treated as bare URLs. Using an `<a>` tag is a legitimate use case when `target="_blank"` is needed, since Markdown link syntax does not support this attribute.

**Comment by shinagawa-web** [comment]
## Root cause analysis

The fix requires rethinking how the scanner handles HTML tag text content.

### Why the current approach falls short

The current `findBareURLs` scanner processes lines character by character and uses `isWrappedURL` to decide whether a URL should be skipped. `isWrappedURL` only looks at what comes **immediately before** the URL (angle bracket, Markdown link syntax, or HTML attribute `="`). It has no concept of HTML element context.

A naive fix — e.g. "skip if preceded by `>` and followed by `</a>`" — is fragile because it conflates any HTML tag's closing `>` with an anchor tag, and says nothing about what comes before the `>`.

### Correct approach

The right fix mirrors what `stripInlineCode` and `stripHTMLComments` already do: **pre-process the line before scanning** by blanking out content that should not be checked.

In CommonMark, inline HTML is part of the document, but the text inside an `<a>` element is a hyperlink by definition — regardless of whether it looks like a bare URL. The scanner should therefore strip the text content of `<a>...</a>` spans (same-line only) before running URL detection, so any URL used as anchor text is never seen by `findBareURLs`.

Concretely, a new `stripAnchorText` function should:
1. Find `<a` (case-insensitive) followed eventually by `>`
2. Find the matching `</a>` on the same line
3. Blank out the text between `>` and `</a>`

This is then called in `prepareScanned` alongside the existing `stripInlineCode` and `stripHTMLComments` steps.

**Comment by shinagawa-web** [comment]
## Resolved

URLs inside HTML `<a href="...">` / `<img src="...">` attributes are no longer flagged as bare URLs. This is covered by the `no-bare-urls` HTML handling and locked in by e2e regression:

- Fixture: `e2e/fixtures/no_bare_urls_valid.md:9` — `<a href="https://example.com"><img src="https://example.com/image.gif" ...></a>`
- Test: `TestE2E_BasicFunctionality/NoBareURLsHTMLAttributeValid` asserts the attribute URLs are not reported.

Part of the #337 shared-context preprocessing work (released in v3.3.0). Closing as fixed.
